src/go/k8s/pkg/resources/statefulset.go

```diff
@@ -269,6 +285,10 @@ func (r *StatefulSetResource) obj(
 		externalAddressType = externalListener.External.PreferredAddressType
 	}
 	tlsVolumes, tlsVolumeMounts := r.volumeProvider.Volumes()
+
+	// We set statefulset replicas via status.replicas in order to control it from the handleScaling function
+	replicas := r.pandaCluster.Status.GetReplicas()
```

Comment:
Nit: I think normally the operator will update the desired state of StatefulSet then this will reconcile back to the owner which is the Cluster.